src/python/pants/engine/README.md

```diff
@@ -27,20 +28,20 @@ produce the requested value.
 ### Products and Params
 
 The engine executes your `@rule`s in order to (recursively) compute a `Product` of the requested
-type for a set of `Param`s. This recursive type search leads to a loosely coupled (and yet
-still statically checked) form of dependency injection.
+type for a set of `Param`s. This recursive type search leads to a loosely coupled (and yet still 
+statically checked) form of dependency injection.
 
-When an `@rule` runs, it requires a set of `Param`s that the engine has determined are needed
-to compute its transitive `@rule` dependencies. So although an `@rule` might not have a particular
+When an `@rule` runs, it requires a set of `Param`s that the engine has determined are needed to
+compute its transitive `@rule` dependencies. So although an `@rule` might not have a particular
 `Param` type in its signature, it might depend on another `@rule` that does need that `Param`, and
 would thus need that `Param` in order to run. To see which `Params` the engine needs to run each
 `@rule`, refer to the `Visualization` section below.
 
 Any hashable type with useful equality may be used as a `Param`, and additional `Params` can be
-provided to an `@rule`'s dependencies via `Get` requests (see below). Each `Param` value in a set
-of `Params` is unique by type, so if `@rules` recursively introduce a particular `Param` type,
-there will still only be one value for that type in each `@rule`, but it will change as you move
-deeper into the dependency graph.
+provided to an `@rule`'s dependencies via `Get` requests (see below). Each `Param` value in a set of
+`Params` is unique by type, so if `@rule`s recursively introduce a particular `Param` type, there
+will still only be one value for that type in each `@rule`, but it will change as you move deeper
+into the dependency graph.
```

Comment:
This is a correct description of what *should* be happening, but there is a bug about this that I believe is still unfixed specifically regarding performing a `yield Get()` with the same subject/product pair twice in the same `@rule` body (where pants fails at runtime). I hadn't yet made an issue about this because I was trying to fix it really quickly, but if I can't do that today I'll make an issue to track this. I believe @illicitonion and @gshuflin have run into this.

Reply:
OK. Thanks in advance for filing. I won't change language here per my out of scope mantra though.